sdk/appcenter/src/main/java/com/microsoft/appcenter/ingestion/OneCollectorIngestion.java

```diff
@@ -0,0 +1,212 @@
+        mLogUrl = DEFAULT_LOG_URL;
+    }
+
+    @Override
+    public ServiceCall sendAsync(String appSecret, UUID installId, LogContainer logContainer, ServiceCallback serviceCallback) throws IllegalArgumentException {
+
+        /* Gather API keys from logs. */
+        Map<String, String> headers = new HashMap<>();
+        Set<String> apiKeys = new LinkedHashSet<>();
+        for (Log log : logContainer.getLogs()) {
+            apiKeys.addAll(log.getTransmissionTargetTokens());
+        }
+
+        /* Build the header. String.join with iterable is only API level 26+. */
+        StringBuilder apiKey = new StringBuilder();
+        for (String targetToken : apiKeys) {
+            apiKey.append(targetToken).append(",");
+        }
+        if (!apiKeys.isEmpty()) {
+            apiKey.deleteCharAt(apiKey.length() - 1);
+        }
+        headers.put(API_KEY, apiKey.toString());
+
+        /* Content type. */
+        headers.put(CONTENT_TYPE_KEY, CONTENT_TYPE_VALUE);
+
+        /* Client version */
+        headers.put(CLIENT_VERSION_KEY, String.format(CLIENT_VERSION_FORMAT, VERSION_NAME));
+
+        /* Content encoding */
+        // TODO Android HttpUrlConnection should take care of this?  Need to verify in fiddler
+
+        /* Upload time */
+        headers.put(UPLOAD_TIME_KEY, String.valueOf(System.currentTimeMillis()));
+
+        /* Make the call. */
+        HttpClient.CallTemplate callTemplate = new IngestionCallTemplate(mLogSerializer, logContainer);
+        return mHttpClient.callAsync(mLogUrl, METHOD_POST, headers, callTemplate, serviceCallback);
+    }
+
+    /**
+     * Update log URL.
+     *
+     * @param logUrl log URL.
+     */
+    @Override
+    @SuppressWarnings("SameParameterValue")
+    public void setLogUrl(@NonNull String logUrl) {
+        mLogUrl = logUrl;
+    }
+
+    @Override
+    public void reopen() {
+        mHttpClient.reopen();
+    }
+
+    @Override
+    public void close() throws IOException {
+        mHttpClient.close();
+    }
+
+    /**
+     * Call template implementation for One Collector.
+     */
+    private static class IngestionCallTemplate implements HttpClient.CallTemplate {
+
+        /**
+         * Log serializer.
+         */
+        private final LogSerializer mLogSerializer;
+
+        /**
+         * Log container.
+         */
+        private final LogContainer mLogContainer;
+
+        /**
+         * Init.
+         */
+        IngestionCallTemplate(LogSerializer logSerializer, LogContainer logContainer) {
+            mLogSerializer = logSerializer;
+            mLogContainer = logContainer;
+        }
+
+        @Override
+        public String buildRequestBody() throws JSONException {
+
+            /* Serialize payload. */
+            StringBuilder jsonStream = new StringBuilder();
+            for (Log log : mLogContainer.getLogs()) {
+                jsonStream.append(mLogSerializer.serializeLog(log));
+                jsonStream.append('\n');
```

Comment:
Not comma (,) separated?
Or, maybe `return mLogSerializer.serializeContainer(mLogContainer);`?

Reply:
serialize container is a different content type, we have to use \n, it's really specific to One Collector.
Once we remove App Center completely we can replace serializeContainer I guess.

Reply:
Then can you add comment to explain this one?

Reply:
I'll add a comment

Reply:
Sure, @jwallra can you comment?